APIs/schemas/transporttype-response-schema.json

```
@@ -3,14 +3,23 @@
   "title": "Transport Type",
   "description": "Transport type URN base used by the Sender or Receiver (i.e. with any subclassifications or versions removed)",
   "type": "string",
-  "oneOf": [
+  "anyOf": [
     {
-      "enum": [
-        "urn:x-nmos:transport:rtp",
-        "urn:x-nmos:transport:dash",
-        "urn:x-nmos:transport:websocket",
-        "urn:x-nmos:transport:mqtt"
+      "oneOf": [
+        {
+          "enum": [
+            "urn:x-nmos:transport:rtp",
+            "urn:x-nmos:transport:dash",
+            "urn:x-nmos:transport:websocket",
+            "urn:x-nmos:transport:mqtt"
+          ]
+        }
       ]
+    },
+    {
+      "description": "Any transport registered in the parameter register",
+      "type": "string",
+      "pattern": "^urn:x-nmos:transport:[a-zA-Z0-9_]+$"
```

Might need a few tweaks to match the description in Use of URNs in IS-04, which defines general NMOS URNs as `<URN-base>[.<subclassification>][/<version>]` and says "the URN base is a sequence of one or more names delimited by `:` characters, beginning with the namespace `urn:x-nmos:`." Of course, unhelpfully, it doesn't go on to define the permitted characters in a "name"... except that it's clear it excludes "." and "/" as well as ":".

The name pattern should allow "-" as well as "_" I think?

So maybe this:

```suggestion
      "pattern": "^urn:x-nmos:transport:[a-zA-Z0-9_-]+(:[a-zA-Z0-9_-]+)*$"
```

If we can think of reasons we might want to allow other values later (e.g. allow subclassifications or versions, or Chinese characters in names), then maybe we'd make it _less_ precise like some other examples in the spec, i.e. just:

```suggestion
      "pattern": "^urn:x-nmos:transport:"
```

---

Something worth debating in an ARG discussion.

---

Worth noting for external parameter keys we use `^ext_[a-zA-Z0-9_]+$`

---

I considered that, but URN are not the same as property names.

---

Noting that IS-04 uses the simpler form everywhere, e.g.

https://github.com/AMWA-TV/is-04/blob/3e7fc65109bf95acfe0d2c30a4e46ee65144684b/APIs/schemas/device.json#L23

https://github.com/AMWA-TV/is-04/blob/3e7fc65109bf95acfe0d2c30a4e46ee65144684b/APIs/schemas/receiver_core.json#L27

However, the difference here is that we want to allow only URN base without subclassification or version.
So another simple formulation could be:

```suggestion
      "pattern": "^urn:x-nmos:transport:[^./]+$"
```